File: 加密保护/WinLicense v2.1.0.10/WinlicenseSDK/ExamplesSDK/Text Keys Registration/Delphi/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, WinlicenseSDK, ExtCtrls;

type
  TForm1 = class(TForm)
    Label1: TLabel;
    Label2: TLabel;
    Button1: TButton;
    GroupBox1: TGroupBox;
    TextKeyEdit: TMemo;
    Panel1: TPanel;
    Label3: TLabel;
    GroupBox2: TGroupBox;
    Label4: TLabel;
    Label5: TLabel;
    Label6: TLabel;
    NameEdit: TEdit;
    CompanyEdit: TEdit;
    ExtradataEdit: TEdit;
    procedure Button1Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.Button1Click(Sender: TObject);
begin

    if WLRegNormalKeyCheck(PChar(TextKeyEdit.Text)) = false  then

        MessageBox(0, 'The key that you entered is invalid. Please, try it again', 'Error', MB_OK or MB_ICONERROR)

    else
    begin

      // if activation is valid, insert the registration info in a file
      // So, winlicense will be able to finish the registration process

      WLRegNormalKeyInstallToFile(PChar(TextKeyEdit.Text));

      MessageBox(0, 'Registration key is correct. Please, restart this application to finish the registration process', 'Success', MB_OK);

    end;

end;

procedure TForm1.FormCreate(Sender: TObject);

var

  Name : ARRAY[0..200] of char;
  Company : ARRAY[0..200] of char;
  ExtraData : ARRAY[0..200] of char;
  ExtStatus: Integer;

begin

   // check if application has been activated (registered)

    if WLRegGetStatus(ExtStatus) = 1 then
    begin

        Panel1.Visible := True;
        WLRegGetLicenseInfo(Name, Company, ExtraData);
        NameEdit.Text := Name;
        CompanyEdit.Text := Company;
        ExtradataEdit.Text := ExtraData;
        
    end;
     
end;

end.

File: 加密保护/WinLicense v2.0.8.0/WinlicenseSDK/ExamplesSDK/SmartActivate Keys/C/CBuilder/Unit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Unit1.h"
#include "..\..\..\..\Include\C\WinlicenseSDK.h"
#pragma comment (lib,"..\\..\\..\\..\\Lib\\OMF\\WinlicenseSDK.lib")

//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
    : TForm(Owner)
{

    // check if application has been activated (registered)

    if (WLRegGetStatus(NULL) == 1)
    {
        Label1->Caption = "This application has been ACTIVATED";
        Label2->Caption = "";
        Button1->Visible = False;
        GroupBox1->Visible = False;
    }

}
//---------------------------------------------------------------------------

void __fastcall TForm1::Button1Click(TObject *Sender)
{

    if (!WLRegSmartKeyCheck(NameEdit->Text.Length()? NameEdit->Text.c_str() : NULL,
         OrganizationEdit->Text.Length()? OrganizationEdit->Text.c_str() : NULL,
         ExtraDataEdit->Text.Length()? ExtraDataEdit->Text.c_str(): NULL, ActivationCodeEdit->Text.c_str()))
    {
        MessageBox(NULL, "Activation code is INVALID", "Error", MB_OK | MB_ICONERROR);
    }
    else
    {
        // if activation is valid, insert the registration info in a file
        // So, winlicense will be able to finish the registration process

        WLRegSmartKeyInstallToFile(NameEdit->Text.c_str(), OrganizationEdit->Text.c_str(), ExtraDataEdit->Text.c_str(), ActivationCodeEdit->Text.c_str());

        MessageBox(NULL, "Activation code is correct. Please, restart\n this application to finish the registration process", "Success", MB_OK);
    }

}
//---------------------------------------------------------------------------
